lib/engine/game/g_18_usa/step/special_buy_train.rb

```diff
@@ -1,12 +1,13 @@
 # frozen_string_literal: true
 
 require_relative '../../../step/special_buy_train'
-
+require_relative 'scrap_train_module'
 module Engine
   module Game
     module G18USA
       module Step
         class SpecialBuyTrain < Engine::Step::SpecialBuyTrain
+          include ScrapTrainModule
```

Comment:
why is this includede in every single step? this smells funny

Reply:
This should be a non-blocking step, like loan.

Reply:
There's some logic that looks at the current step for the methods in `ScrapTrainModule`, I can look into if making that logic search for a ScrapTrains step instead of the current step would work

Reply:
Yeah, let's rework the view to find the step. For all other games, it's likely the buy train step, so active step == scrap train step for them.

Reply:
We'll just have to test those games out to verify.

Reply:
Using `        step = @game.round.step_for(@corporation, 'scrap_train')` instead of `@game.round.active_step` worked well as a replacement, I can test 18Mag but I'm not sure how to play some of the other games like 1840